packages/slate-editor/src/extensions/allowed-blocks/AllowedBlocksExtension.ts

```diff
@@ -0,0 +1,42 @@
+import type { Extension } from '@prezly/slate-commons';
+import { isElementNode } from '@prezly/slate-types';
+import { get } from 'lodash-es';
+import { type Editor, type NodeEntry, Transforms } from 'slate';
+
+import type { AllowedBlocksExtensionConfiguration, Block } from './types';
+
+export const EXTENSION_ID = 'AllowedBlocksExtension';
+
+export function AllowedBlocksExtension({ blocks }: AllowedBlocksExtensionConfiguration): Extension {
+    return {
+        id: EXTENSION_ID,
+        normalizeNode(editor: Editor, [node, path]: NodeEntry) {
```

Comment:
Here's an [example](https://github.com/prezly/prezly/pull/11617/files#diff-f9a6d5f0592725fbbd69b38ca49032ddafed7e0bf91f222668b4d6ec5897d169R42-R53) on how it's used.

Each entry in the `blocks` array corresponds to the top-level path (or row if you want).
For the current `path`, we check whether the current node is one of the types specified in `types` array and if it's not, it will be replaced by a block returned from `defaultBlockFn`.